drivers/counter/counter_nrfx_rtc.c

```diff
@@ -68,226 +74,428 @@ static inline const struct counter_nrfx_config *get_nrfx_config(
 				struct counter_nrfx_config, info);
 }
 
-static int counter_nrfx_start(struct device *dev)
+static int start(struct device *dev)
 {
-	nrfx_rtc_enable(&get_nrfx_config(dev)->rtc);
+	nrf_rtc_task_trigger(get_nrfx_config(dev)->rtc, NRF_RTC_TASK_START);
 
 	return 0;
 }
 
-static int counter_nrfx_stop(struct device *dev)
+static int stop(struct device *dev)
 {
-	nrfx_rtc_disable(&get_nrfx_config(dev)->rtc);
+	nrf_rtc_task_trigger(get_nrfx_config(dev)->rtc, NRF_RTC_TASK_STOP);
+	nrf_rtc_task_trigger(get_nrfx_config(dev)->rtc, NRF_RTC_TASK_CLEAR);
```

Comment:
Why `CLEAR`? The counter API does not say that the counter should be cleared after it is stopped.

Reply:
on the other hand it does not say that it shouldn't 😄. If we assume that counter is not cleared after stop and not cleared when started then counter API is missing `counter_clear` call. On the other hand for some implementations it might be hard to maintain  "no cleaning" approach. For example in NRF_TIMER if `TASKS_SHUTDOWN` is used to stop then counter is reset.

I think that it would be better to add `counter_pause` (some may return not supported) in a future and specify that `stop` clears the counter.

Reply:
My comment was not towards preserving the current value while the counter is stopped, as the API does not specify that counters should provide that. The thing is that everything should work without this clearing. And it was, with the previous implementation using the `nrfx_rtc` driver. Thus, if this clearing is needed now, it may indicate that something (a test?) is relying on a non-specified behavior.

Reply:
removed clearing. test is passing.